### PR TITLE
Allow removing asset from embedded asset registry

### PR DESCRIPTION
**Pull request by extrawurst**
# Objective

- Allow not only inserting `Data` into `EmbeddedAssetRegistry` and `Dir` in turn but now also removing it again.
- This way when used to embed asset data from *somewhere* but not load it using the conventional means via `AssetServer` (which I observed takes ownership of the `Data`) the `Data` does not need to stay in memory of the `EmbeddedAssetRegistry` throughout the lifetime of the application.

## Solution

- added the `remove_asset` functions in `EmbeddedAssetRegistry` and `Dir`

## Testing

- added a module unittest
- does this require changes if build with feature `embedded_watcher`?

### CODE REVIEW

File: crates/bevy_asset/src/io/memory.rs [hunk 1]
```diff
@@ -55,6 +55,16 @@ impl Dir {
         );
     }
 
+    pub fn remove_asset(&self, path: &Path) -> Option<Data> {
```
Comment (alice-i-cecile) [inline crates/bevy_asset/src/io/memory.rs:58]: Doc strings here too :)

Comment (extrawurst) [reply crates/bevy_asset/src/io/memory.rs:58]: Wanted to keep it consistent 😂

Comment (alice-i-cecile) [reply crates/bevy_asset/src/io/memory.rs:58]: Yeah but our current lack of docs in the siblings is *bad*!

Comment (extrawurst) [reply crates/bevy_asset/src/io/memory.rs:58]: I was at least 50% kidding ;) will add them

File: crates/bevy_asset/src/io/embedded/mod.rs [hunk 2]
```diff
@@ -404,4 +411,13 @@ mod tests {
         // Really, should be "my_crate/src/the/asset.png"
         assert_eq!(asset_path, Path::new("my_crate/the/asset.png"));
     }
+
+    #[test]
+    fn remove_embedded_asset() {
```
Comment (NiseVoid) [inline crates/bevy_asset/src/io/embedded/mod.rs:416]: Could this test be expanded to ensure the asset is now returned as not existing? The test currently only checks that you can call `remove_asset` once, but not that it actually removes anything

**Review by MScottMcBee — APPROVED** [review]

**Review by NiseVoid — APPROVED** [review]
Seems about right overall, one minor nit on the test

File: crates/bevy_asset/src/io/embedded/mod.rs (hunk 2 above)
Comment (extrawurst) [reply crates/bevy_asset/src/io/embedded/mod.rs:416]: ok i added asserts for that

**Review by VALEK9292 — COMMENTED** [review]
Ft

**Pull request merged**